File: Python/4. 큐/2161.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/2161
'''

n = int(input())

l = [i for i in range(1, n+1)]

while len(l) > 1 :
    print(l[0], end = ' ')
    l.pop(0)
    l.append(l.pop(0))
print(l[0])

File: C++/10. 큐 - 초급/2161.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/2161
*/

#include <iostream>
#include <queue>

using namespace std;

int n;
queue<int> q;

int main(){
    
    cin >> n;
    for(int i=1; i<=n; i++){
        q.push(i);
    }

    while(!q.empty()){
        cout << q.front() << " ";
        q.pop();
        q.push(q.front());
        q.pop();
    }
    
    return 0;
}